### ISSUE TITLE
Merge Request CI status not updating

### ISSUE THREAD

**Issue by ndcampbell** [DABSquared/gitlab-plugin#56]
On a Gitlab merge request the CI status stays at "CI build pending for 6f3ca78b. Build page". The Build page link does not go to the build status (just goes to a blank Jenkins page) and even when the build is successful this pending message is never changed.

**Comment by bassrock** [comment]
Can you post any logs from Jenkins?

**Comment by mildewonrice** [comment]
Here are some of the logs I have from the jenkins.log. This is when one of my builds kicked off from a Merge Request and failed. The CI Status in Gitlab never gets updated at all. Let me know if there is anything else I can provide. 

Feb 11, 2015 9:58:50 AM com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WARNING: WebHook called.
Feb 11, 2015 9:58:54 AM com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WARNING: WebHook called.
Feb 11, 2015 9:58:56 AM com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WARNING: WebHook called.
Feb 11, 2015 9:59:21 AM com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WARNING: WebHook called.
Feb 11, 2015 9:59:39 AM hudson.model.Run execute
INFO: Venus Git Build #26 main build action completed: FAILURE
Feb 11, 2015 9:59:40 AM org.terracotta.jenkins.plugins.postcompleted.PostCompletedRunListener onCompleted

**Comment by ejhayes** [comment]
I'm running into the same issue as well:

`https://****/project/XXXX/commits/047c93e28773ad2122ff16bd0fae9574ae78995b`

Returns nothing

`https://****/project/XXXX/builds/047c93e28773ad2122ff16bd0fae9574ae78995b/status.json`

Returns:

`{
sha: "047c93e28773ad2122ff16bd0fae9574ae78995b",
status: "pending"
}`

**Comment by bassrock** [comment]
@ejhayes  and @mildewonrice what version of Gitlab?

**Comment by ejhayes** [comment]
I'm using GitLab 7.6.2

**Comment by mildewonrice** [comment]
I am running 7.7.2-ee

**Comment by Kurry** [comment]
This seems like more of a plugin issue rather than a gitlab issue:

1) The HTTP redirect from the SHA1 to build page never occurs 

2) After the fails or completes the status.json file does not update.

**Comment by bhaku** [comment]
This same problem. GitLab 7.5.3, Jenkins 1.580.2 using Gitlab-plugin 1.1.15

**Comment by twk3** [comment]
I have the same problem on Gitlab 7.8.4, Jenkins 1.580.3 using Gitlab-plugin 1.1.15
Git Plugin: 2.3.5 Git client plugin: 1.16.1

status.json returns pending for completed builds, and build page is blank

**Comment by bhaku** [comment]
Upgrade to version 1.1.17 resolve issue for me.

**Comment by twk3** [comment]
Still a problem for me on 1.1.17

**Comment by ndcampbell** [comment]
1.1.17 fixed this problem for me on Gitlab 7.8.1-ee.

**Comment by bassrock** [comment]
@twk3 and @ndcampbell is this for a merge request between forks or between branches of the same repos? Its possible that one works while the other does not.

**Comment by bassrock** [comment]
And could be related to #61

**Comment by twk3** [comment]
For between branches of the same repo for me. But the repo itself is a fork.

**Comment by bassrock** [comment]
@twk3 can you post the logs while it is requesting the status. There was more logging added in the lastest release.

**Comment by twk3** [comment]
Log level is set at INFO, all I get for a build status call is
```
Apr 14, 2015 11:10:18 AM WARNING com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WebHook called.
```

For pushing to to a merge request branch I get:
```
Apr 14, 2015 11:13:42 AM WARNING com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WebHook called.
Apr 14, 2015 11:13:42 AM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$1 run
p4gitlab-test-1 triggered for push.
Apr 14, 2015 11:13:42 AM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$1 createActions
GitLab Push Request from branch test-jenkins.
Apr 14, 2015 11:13:42 AM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$1 createActions
Trying to get name and URL for job: p4gitlab-test-1 using project p4gitlab-test-1 (push)
Apr 14, 2015 11:13:42 AM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$1 run
GitLab Push Request detected in p4gitlab-test-1. Triggering  #16
```

**Comment by twk3** [comment]
Updated to latest plugin, and changed log level to fine. The get Status log didn't change. Doing a push to a merge results in:
```
Apr 14, 2015 11:25:29 AM WARNING com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WebHook called.
Apr 14, 2015 11:25:29 AM FINE com.dabsquared.gitlabjenkins.GitLabWebHook generateBuild
data: {
    "object_kind": "push",
    "before": "17c8d6cd54ee7a510aca8f101bf0f8beaeb39ba9",
    "after": "2a3c05cad9e3e1db66bc6dc8b4c3c6e5c893e847",
    "ref": "refs/heads/test-jenkins",
    "checkout_sha": "2a3c05cad9e3e1db66bc6dc8b4c3c6e5c893e847",
    "user_id": 19,
    "user_name": "[username]",
    "user_email": "[email]",
    "project_id": 91,
    "repository":     {
        "name": "gitlab-shell",
        "url": "git@gitlab.example.com:username/gitlab-shell.git",
        "description": "SSH access and repository management app for GitLab\r\n",
        "homepage": "http://gitlab.example.com/username/gitlab-shell",
        "git_http_url": "http://gitlab.example.com/username/gitlab-shell.git",
        "git_ssh_url": "git@gitlab.example.com:username/gitlab-shell.git",
        "visibility_level": 20
    },
    "commits": [    {
        "id": "2a3c05cad9e3e1db66bc6dc8b4c3c6e5c893e847",
        "message": "test\n",
        "timestamp": "2015-04-14T11:25:25-07:00",
        "url": "http://gitlab.example.com/username/gitlab-shell/commit/2a3c05cad9e3e1db66bc6dc8b4c3c6e5c893e847",
        "author":         {
            "name": "[name]",
            "email": "[email]"
        }
    }],
    "total_commits_count": 1
}
Apr 14, 2015 11:25:29 AM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$1 run
p4gitlab-test-1 triggered for push.
Apr 14, 2015 11:25:29 AM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$1 createActions
GitLab Push Request from branch test-jenkins.
Apr 14, 2015 11:25:29 AM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$1 createActions
Trying to get name and URL for job: p4gitlab-test-1 using project p4gitlab-test-1 (push)
Apr 14, 2015 11:25:29 AM INFO com.dabsquared.gitlabjenkins.GitLabPushTrigger$1 run
GitLab Push Request detected in p4gitlab-test-1. Triggering  #17
```
(username and url changed)

**Comment by bassrock** [comment]
If you go to the url:

/project/PROJECT_NAME/builds/COMMIT_SHA1/status.json

with your project name in jenkins and "2a3c05cad9e3e1db66bc6dc8b4c3c6e5c893e847" from your log above what do you get

**Comment by bassrock** [comment]
On the jenkins box^

**Comment by twk3** [comment]
The log says
```
Apr 14, 2015 11:37:03 AM WARNING com.dabsquared.gitlabjenkins.GitLabWebHook getDynamic
WebHook called.
```
and the response is:
```
{"sha":"2a3c05cad9e3e1db66bc6dc8b4c3c6e5c893e847","status":"pending"}
```

**Comment by bassrock** [comment]
Ok and i would assume the merge request page shows a pending status?

**Comment by twk3** [comment]
Yep, but build is finished, and successful

**Comment by bassrock** [comment]
Ok So this function in GitlabWebHook:

https://github.com/DABSquared/gitlab-plugin/blob/master/src/main/java/com/dabsquared/gitlabjenkins/GitLabWebHook.java#L456

is not returning a build from the commit sha1.

**Comment by b4oshany** [comment]
I'm having the same issue as well.

**Comment by ndcampbell** [comment]
This issue is happening again for me on Gitlab 7.10 for branch merges. I am using the latest version of the plugin. CI status is stuck at pending. It was working for me just fine in Gitlab 7.8

**Comment by djb61** [comment]
@ndcampbell This is broken when using Gitlab 7.10 as Gitlab has changed the CI integration API. See #74 for details.

**Comment by bassrock** [comment]
@ndcampbell this should be fixed in the latest release. Can you confirm?

**Comment by ndcampbell** [comment]
@bassrock yeah, I can confirm. Looks to be working great in 7.10 with the latest release.

**Comment by bassrock** [comment]
good to hear

**Comment by twk3** [comment]
I still have this issue on the latest plugin and gitlab 7.10.

The older 1.1.3 plugin is still working

I will try and debug at some point, and find why GitlabWebHook doesn't return a build in my case.

**Comment by marciobueno1** [comment]
Only the older version (1.1.3) of this plugin is working with the GitLab 7.2.1.

To build from all branches (including new ones), I am using the polling method triggered by a Web Hook in GitLab, using a URL like this: http://JENKINS_URL/git/notifyCommit?url=URL_GIT_REPOSITORY

With new new plugin version, the status page on Merge Request in GitLab is showing "pending", when I downgraded to 1.1.3, it started to show "passed". Just to reinforce, no other configuration was changed, only did the downgrade of the plugin.
